Sources/HAP/Base/Service.swift

```diff
@@ -85,28 +85,45 @@ public extension ServiceType {
     static let garageDoorOpener = ServiceType(0x41)
     static let lockMechanism = ServiceType(0x45)
     static let securitySystem = ServiceType(0x7E)
+    // Television support
+    static let television = ServiceType(0xD8)
+    static let inputSource = ServiceType(0xD9)
+    static let televisionSpeaker = ServiceType(0x113)
 }
 
-open class Service: JSONSerializable {
+open class Service: NSObject, JSONSerializable {
```

Comment:
Why inherit from NSObject? In order to support Linux, we probably cannot do this.

Reply:
Extending NSObject simply brings in Hashable conformity for free. Hashable conformity is required by WeakObjectSet.

I note that both Server and Server.Connection both already extend NSObject. The first case is in order for the Server to be a NetServiceDelegate. The second is also to use Server.Connections in a WeakObjectSet.

If HAP currently works under linux, then it should be OK to make Service extend NSObject too.